packages/@expo/server/src/index.ts

```diff
@@ -152,6 +147,42 @@ export function createRequestHandler(
 
     debug('Request', sanitizedPathname);
 
+    if (routesManifest.rewrites) {
+      for (const route of routesManifest.rewrites) {
+        if (!route.namedRegex.test(sanitizedPathname)) {
+          continue;
+        }
+
+        const url = getRedirectRewriteLocation(request, route);
+
+        if (url) {
+          request = new Request(new URL(url, new URL(request.url).origin), request);
```

Comment:
This won't affect `sanitizedPathname` going forward, so may be ineffective, if I'm not mistaken. Noticed this because this is written differently internally for hosting.